fw/cache.c

```diff
@@ -2265,10 +2285,18 @@ tfw_cache_do_action(TfwHttpMsg *msg, tfw_http_cache_cb_t action)
 	}
 
 	req = (TfwHttpReq *)msg;
-	if (unlikely(req->method == TFW_HTTP_METH_PURGE))
-		tfw_cache_purge_method(req);
-	else
+	if (unlikely(req->method == TFW_HTTP_METH_PURGE)) {
+		if (tfw_cache_purge_method(req))
+			return;
+		if (unlikely(req->cache_ctl.flags & TFW_HTTP_CC_CACHE_PURGE)) {
+			cache_req_process_node(req, action);
```

Comment:
This call doesn't look good since the function tries to fetch the TDB cache entry and should not find it since it's already PURGEed and only then we go to the `action()` callback. Why not to call the callback right here?